c2rust-transpile/src/c_ast/mod.rs

```diff
@@ -1865,3 +1870,247 @@ impl CTypeKind {
+                line: 7,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 9,
+                column: 7,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 9,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 9,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 5,
+                column: 7,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 5,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 5,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 8,
+                column: 3,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 8,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 7,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 0,
+                column: 4,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 0,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 2,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 98,
+                column: 3,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 98,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 202,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 5,
+                line: 202,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 7,
+                line: 1,
+                column: 1,
+            },
+            SrcLoc {
+                fileid: 7,
+                line: 3,
+                column: 1,
+            },
+        ];
+
+        // This should not panic
+        locs.sort_unstable_by(|a, b| ctx.compare_src_locs(a, b));
+
+        let n = locs.len();
+        for i in 0..n {
+            let a = locs[i];
+            for j in 0..n {
+                let b = locs[j];
+                for k in 0..n {
+                    let c = locs[k];
+                    if a < b && b < c {
+                        assert!(a < c, "Total order has been violated");
+                    }
+                    if a > b && b > c {
+                        assert!(a > c, "Total order has been violated");
+                    }
+                    if a == b && b == c {
+                        assert_eq!(a, c, "Total order has been violated");
+                    }
```

Comment:
We need to use `compare_src_locs` here, not the `impl Ord`.

```suggestion
                    let ab = a.cmp(b);
                    let bc = b.cmp(c);
                    let ac = a.cmp(c);
                    if ab == bc {
                        assert_eq!(ab, ac, "Total order (transivitity) has been violated");
                    }
```